log/log.go

```diff
@@ -1,72 +1,129 @@
 package log
 
 import (
-	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
 
-	"github.com/golang/glog"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
 )
 
-const (
-	debug glog.Level = glog.Level(4)
-	trace glog.Level = glog.Level(5)
-)
+// Configures the global log level and log format.
+func InitializeLogger() zerolog.Logger {
+	logTimeFieldFormat, isTimeFieldFormatDefined := os.LookupEnv("LOG_TIME_FIELD_FORMAT")
+
+	if !isTimeFieldFormatDefined {
+		logTimeFieldFormat = time.RFC3339
+	}
+	zerolog.TimeFieldFormat = logTimeFieldFormat
+
+	logSamplerRateAsString, isSamplerRateDefined := os.LookupEnv("LOG_SAMPLER_RATE")
+	if isSamplerRateDefined {
+		logSamplerRate, err := strconv.Atoi(logSamplerRateAsString)
+		if err != nil {
+			log.Warn().Msgf("Provided sampling rate %s cannot be parsed to int32. "+
+				"No sampling rate will be set. Error: %v", logSamplerRateAsString, err)
+		} else {
+			log.Debug().Msgf("Setting log sample rate to every %dth event", logSamplerRate)
+			log.Logger = log.Sample(&zerolog.BasicSampler{N: uint32(logSamplerRate)})
+		}
+	}
+
+	if os.Getenv("LOG_FORMAT") != "json" {
+		log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stdout, TimeFormat: zerolog.TimeFieldFormat})
+	}
```

Comment:
Again, I wonder if we should test for validity of the LOG_FORMAT using some zerolog parse function?

Reply:
Guess this is already some kind of validation. Everything else than `json` will result in a regular text output. We could validate against an enum of text and json to give some debugging info to the user. Guess that might be reasonable to do.